### ISSUE TITLE
Ray actor child processes still running after ray.kill()

### ISSUE THREAD

**Issue by PidgeyBE** [ray-project/ray#11108]
### What is the problem?

* ray 0.8.7 and 1.0.0rc2
* When a subprocess is created in a ray actor, the subprocess is not shut down when the actor is killed.
* This is in particular an issue when using keras fit https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit with `use_multiprocessing=True`
* related to https://github.com/ray-project/ray/pull/3297

### Reproduction (REQUIRED)

```
import os
import subprocess
import time
import ray

ray.init()

UNIQUE_CMD = "sleep {}".format(str(time.time()))

@ray.remote
class Counter():
    def __init__(self):
        self.counter = 0
        self.subproc = subprocess.Popen(UNIQUE_CMD, shell=True)

    def kill(self):
        ray.actor.exit_actor()


c = Counter.remote()
time.sleep(5)  # sleep to let the actor initialize
c.kill.remote()  # first way to kill
ray.kill(c)  # second method to kill
ray.shutdown()  # should also be removed after a shutdown
```
When executing this script, you can see via `htop` that a `sleep` process is left behind.
Is this intended behaviour?

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/latest/installation.html).

BR, Pieterjan

**Comment by rkooo567** [comment]
When I run it without ray, it looks like it has the same behavior? 

```
import os
import subprocess
import time
UNIQUE_CMD = "sleep {}".format(str(time.time()))
subproc = subprocess.Popen(UNIQUE_CMD, shell=True)
time.sleep(5)
```
And when the script is done, the subprocess is still alive. I assume it is just the default behavior of subproces?

**Comment by PidgeyBE** [comment]
Thanks for your answer @rkooo567 . 
In the reproduction script I'm indeed using subprocess, but we stumbled upon this issue when trying to stop a machine learning training that was running in Keras and using multiprocessing (see initial post). When the actor that is running the training is killed, the training just continues and the GPU memory is not released, which is a major issue.
Also I found this ticket https://github.com/ray-project/ray/pull/3297 from which I interpret that child process should indeed share fate with their parent, but maybe I'm interpreting this wrong?
